### ISSUE TITLE
Error on installing gem RACC with image "redmine:4.2-passenger"

### ISSUE THREAD

**Issue by paolobattistella** [docker-library/redmine#282]
I'm trying to start a docker container with image "redmine:4.2-passenger"
I have an issue on installing gem **RACC**.
Do you have any suggestions?

The logs are following:

> 2023-01-03 10:04:49 Your Gemfile lists the gem redmine_crm (>= 0) more than once.
> 2023-01-03 10:04:49 You should probably keep only one of them.
> 2023-01-03 10:04:49 Remove any duplicate entries and specify the gem only once.
> 2023-01-03 10:04:49 While it's not a problem now, it could cause errors if you change the version of one of them later.
> 2023-01-03 10:04:49 Bundler can't satisfy your Gemfile's dependencies.
> 2023-01-03 10:04:49 Install missing gems with `bundle install`.
> 2023-01-03 10:04:50 Your Gemfile lists the gem redmine_crm (>= 0) more than once.
> 2023-01-03 10:04:50 You should probably keep only one of them.
> 2023-01-03 10:04:50 Remove any duplicate entries and specify the gem only once.
> 2023-01-03 10:04:50 While it's not a problem now, it could cause errors if you change the version of one of them later.
> 2023-01-03 10:04:50 The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x64-mingw32, x86-mswin32. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x64-mingw32 x86-mswin32`.
> 2023-01-03 10:04:50 The dependency ffi (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x64-mingw32, x86-mswin32. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x64-mingw32 x86-mswin32`.
> 2023-01-03 10:04:51 Fetching gem metadata from https://rubygems.org/..........
> 2023-01-03 10:04:51 Fetching gem metadata from https://rubygems.org/.
> 2023-01-03 10:04:52 Resolving dependencies....
> 2023-01-03 10:04:52 Using rake 13.0.6
> 2023-01-03 10:04:52 Using concurrent-ruby 1.1.10
> 2023-01-03 10:04:52 Using minitest 5.17.0
> 2023-01-03 10:04:52 Using thread_safe 0.3.6
> 2023-01-03 10:04:52 Using builder 3.2.4
> 2023-01-03 10:04:52 Using erubi 1.12.0
> 2023-01-03 10:04:52 Using mini_portile2 2.5.3
> 2023-01-03 10:04:52 Using crass 1.0.6
> 2023-01-03 10:04:52 Fetching racc 1.6.2
> 2023-01-03 10:04:52 Using rack 2.2.5
> 2023-01-03 10:04:52 Using nio4r 2.5.8
> 2023-01-03 10:04:52 Using websocket-extensions 0.1.5
> 2023-01-03 10:04:52 Using mini_mime 1.0.3
> 2023-01-03 10:04:52 Using method_source 1.0.0
> 2023-01-03 10:04:52 Using thor 1.2.1
> 2023-01-03 10:04:52 Using arel 9.0.0
> 2023-01-03 10:04:52 Using marcel 1.0.2
> 2023-01-03 10:04:52 Using public_suffix 5.0.1
> 2023-01-03 10:04:52 Using bundler 2.1.4
> 2023-01-03 10:04:52 Using chunky_png 1.4.0
> 2023-01-03 10:04:52 Using csv 3.1.9
> 2023-01-03 10:04:52 Using temple 0.9.1
> 2023-01-03 10:04:52 Using tilt 2.0.11
> 2023-01-03 10:04:52 Using htmlentities 4.3.4
> 2023-01-03 10:04:52 Using liquid 2.6.3
> 2023-01-03 10:04:52 Using mini_magick 4.11.0
> 2023-01-03 10:04:52 Using mysql2 0.5.4
> 2023-01-03 10:04:52 Using net-ldap 0.17.1
> 2023-01-03 10:04:52 Using ruby-openid 2.9.2
> 2023-01-03 10:04:52 Using rbpdf-font 1.19.1
> 2023-01-03 10:04:52 Using redcarpet 3.5.1
> 2023-01-03 10:04:52 Using rubyzip 2.3.2
> 2023-01-03 10:04:52 Using rotp 6.2.2
> 2023-01-03 10:04:52 Using rouge 3.26.1
> 2023-01-03 10:04:52 Using rqrcode_core 1.2.0
> 2023-01-03 10:04:52 Using i18n 1.8.11
> 2023-01-03 10:04:52 Using rack-test 2.0.2
> 2023-01-03 10:04:52 Using sprockets 4.2.0
> 2023-01-03 10:04:52 Using request_store 1.5.1
> 2023-01-03 10:04:52 Using websocket-driver 0.7.5
> 2023-01-03 10:04:52 Using mail 2.7.1
> 2023-01-03 10:04:52 Using tzinfo 1.2.10
> 2023-01-03 10:04:52 Using addressable 2.8.1
> 2023-01-03 10:04:52 Fetching haml 6.1.1
> 2023-01-03 10:04:52 Installing haml 6.1.1 with native extensions
> 2023-01-03 10:04:52 Using activesupport 5.2.6.2
> 2023-01-03 10:04:52 Using css_parser 1.13.0
> 2023-01-03 10:04:52 Using rbpdf 1.20.1
> 2023-01-03 10:04:52 Using rqrcode 2.1.2
> 2023-01-03 10:04:52 Using globalid 1.0.0
> 2023-01-03 10:04:52 Using activemodel 5.2.6.2
> 2023-01-03 10:04:52 Using rack-openid 1.4.2
> 2023-01-03 10:04:52 Using activejob 5.2.6.2
> 2023-01-03 10:04:52 Using activerecord 5.2.6.2
> 2023-01-03 10:04:52 Installing racc 1.6.2 with native extensions
> 2023-01-03 10:04:52 Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52     current directory: /usr/local/bundle/gems/racc-1.6.2/ext/racc/cparse
> 2023-01-03 10:04:52 /usr/local/bin/ruby -I /usr/local/lib/ruby/2.7.0 -r
> 2023-01-03 10:04:52 ./siteconf20230103-24-1gyq094.rb extconf.rb
> 2023-01-03 10:04:52 checking for rb_block_call()... *** extconf.rb failed ***
> 2023-01-03 10:04:52 Could not create Makefile due to some reason, probably lack of necessary
> 2023-01-03 10:04:52 libraries and/or headers.  Check the mkmf.log file for more details.  You may
> 2023-01-03 10:04:52 need configuration options.
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 Provided configuration options:
> 2023-01-03 10:04:52 --with-opt-dir
> 2023-01-03 10:04:52 --without-opt-dir
> 2023-01-03 10:04:52 --with-opt-include
> 2023-01-03 10:04:52 --without-opt-include=${opt-dir}/include
> 2023-01-03 10:04:52 --with-opt-lib
> 2023-01-03 10:04:52 --without-opt-lib=${opt-dir}/lib
> 2023-01-03 10:04:52 --with-make-prog
> 2023-01-03 10:04:52 --without-make-prog
> 2023-01-03 10:04:52 --srcdir=.
> 2023-01-03 10:04:52 --curdir
> 2023-01-03 10:04:52 --ruby=/usr/local/bin/$(RUBY_BASE_NAME)
> 2023-01-03 10:04:52 /usr/local/lib/ruby/2.7.0/mkmf.rb:471:in `try_do': The compiler failed to
> 2023-01-03 10:04:52 generate an executable file. (RuntimeError)
> 2023-01-03 10:04:52 You have to install development tools first.
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:564:in `try_link0'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:582:in `try_link'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:794:in `try_func'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:1083:in `block in have_func'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:971:in `block in checking_for'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:361:in `block (2 levels) in postpone'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:331:in `open'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:361:in `block in postpone'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:331:in `open'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:357:in `postpone'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:970:in `checking_for'
> 2023-01-03 10:04:52 from /usr/local/lib/ruby/2.7.0/mkmf.rb:1082:in `have_func'
> 2023-01-03 10:04:52 from extconf.rb:6:in `<main>'
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 To see why this extension failed to compile, please check the mkmf.log which can
> 2023-01-03 10:04:52 be found here:
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52   /usr/local/bundle/extensions/x86_64-linux/2.7.0/racc-1.6.2/mkmf.log
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 extconf failed, exit code 1
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 Gem files will remain installed in /usr/local/bundle/gems/racc-1.6.2 for
> 2023-01-03 10:04:52 inspection.
> 2023-01-03 10:04:52 Results logged to
> 2023-01-03 10:04:52 /usr/local/bundle/extensions/x86_64-linux/2.7.0/racc-1.6.2/gem_make.out
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 An error occurred while installing racc (1.6.2), and Bundler cannot continue.
> 2023-01-03 10:04:52 Make sure that `gem install racc -v '1.6.2' --source 'https://rubygems.org/'`
> 2023-01-03 10:04:52 succeeds before bundling.
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 In Gemfile:
> 2023-01-03 10:04:52   redmine_crm was resolved to 0.0.58, which depends on
> 2023-01-03 10:04:52     rails was resolved to 5.2.6.2, which depends on
> 2023-01-03 10:04:52       actioncable was resolved to 5.2.6.2, which depends on
> 2023-01-03 10:04:52         actionpack was resolved to 5.2.6.2, which depends on
> 2023-01-03 10:04:52           actionview was resolved to 5.2.6.2, which depends on
> 2023-01-03 10:04:52             rails-dom-testing was resolved to 2.0.3, which depends on
> 2023-01-03 10:04:52               nokogiri was resolved to 1.11.7, which depends on
> 2023-01-03 10:04:52                 racc
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52     current directory: /usr/local/bundle/gems/haml-6.1.1/ext/haml
> 2023-01-03 10:04:52 /usr/local/bin/ruby -I /usr/local/lib/ruby/2.7.0 -r
> 2023-01-03 10:04:52 ./siteconf20230103-24-jdoytj.rb extconf.rb
> 2023-01-03 10:04:52 creating Makefile
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 current directory: /usr/local/bundle/gems/haml-6.1.1/ext/haml
> 2023-01-03 10:04:52 make "DESTDIR=" clean
> 2023-01-03 10:04:52 sh: 1: make: not found
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 current directory: /usr/local/bundle/gems/haml-6.1.1/ext/haml
> 2023-01-03 10:04:52 make "DESTDIR="
> 2023-01-03 10:04:52 sh: 1: make: not found
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 make failed, exit code 127
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 Gem files will remain installed in /usr/local/bundle/gems/haml-6.1.1 for
> 2023-01-03 10:04:52 inspection.
> 2023-01-03 10:04:52 Results logged to
> 2023-01-03 10:04:52 /usr/local/bundle/extensions/x86_64-linux/2.7.0/haml-6.1.1/gem_make.out
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 An error occurred while installing haml (6.1.1), and Bundler cannot continue.
> 2023-01-03 10:04:52 Make sure that `gem install haml -v '6.1.1' --source 'https://rubygems.org/'`
> 2023-01-03 10:04:52 succeeds before bundling.
> 2023-01-03 10:04:52 
> 2023-01-03 10:04:52 In Gemfile:
> 2023-01-03 10:04:52   haml-rails was resolved to 2.1.0, which depends on
> 2023-01-03 10:04:52     haml

**Comment by tianon** [comment]
These sorts of questions/requests would be more appropriately posted to a dedicated support forum, such as [the Docker Community Slack](https://dockr.ly/comm-slack), [Server Fault](https://serverfault.com/help/on-topic), [Unix & Linux](https://unix.stackexchange.com/help/on-topic), or [Stack Overflow](https://stackoverflow.com/help/on-topic).
